src/admin/UserForm.js

```diff
@@ -0,0 +1,114 @@
+import {
+  Grid, TextField, Button, InputLabel,
+} from '@material-ui/core';
+import Select from 'react-select';
+import React from 'react';
+import { useQuery, useMutation } from '@apollo/client';
+import useStyles from './useStyles';
+import { GET_USER_ROLES, CREATE_USER, UPDATE_USER } from '../queries/UserQueries';
+import Info from '../Info';
+
+export const validateUser = (
+  username, firstName, lastName, password, userRole, callback,
+) => (
+  callback(!!(username && firstName && lastName && password && userRole)));
+
+function UserForm({ user, callApi, errorAutoHide = 2500 }) {
+  const [username, setUsername] = React.useState(user?.username || '');
+  const [firstName, setFirstName] = React.useState(user?.firstName || '');
+  const [lastName, setLastName] = React.useState(user?.lastName || '');
+  const [password, setPassword] = React.useState(user?.password || '');
+  const [userRole, setUserRole] = React.useState(user?.userRole ? user.userRole : '');
+  const [showError, setShowError] = React.useState(false);
+  const [valid, setValid] = React.useState(!!user);
+  const { data, loading } = useQuery(GET_USER_ROLES);
+  const [createUser] = useMutation(CREATE_USER);
+  const [updateUser] = useMutation(UPDATE_USER);
+  const classes = useStyles();
+
+  React.useEffect(() => {
+    validateUser(username, firstName, lastName, password, userRole, setValid);
+  }, [username, firstName, lastName, password, userRole]);
+
+  const memoUserRoles = React.useMemo(() => (
+    data?.userRoles.map((ur) => ({ value: ur, label: ur.name }))
+  ), [data]);
+
+  const onClick = async () => {
+    const variables = {
+      username, firstName, lastName, password, userRoleId: userRole.id,
+    };
+    try {
+      if (user) {
```

Comment:
`callApi(updateUser, { ...variables, id: user?.id });` should be enough, `id` would become `undefined` and would behave as expected